benchmark-framework/src/main/java/io/openmessaging/benchmark/Benchmark.java

```diff
@@ -192,6 +195,7 @@ public static void main(String[] args) throws Exception {
 
                         generator.close();
                     } catch (Exception e) {
+                        anyFailed.set(true);
```

Comment:
This shouldn't be needed as the program exits at the end of this `catch` block with a non-zero exit code anyway.

Reply:
At line 202

Reply:
Oh interesting, this came in as part of Denis's changes after I made this change: I guess he ran into the same thing and used this approach. I am not so sure about this exit(1) apporach b/c without stopping the workers they will just keep churning away indefinite on their existing workload, right? 

I think `stopAll` does eventually terminate but it might take like 300 seconds or something for the `post` to time out which is pretty annoying. Maybe we should do a stopAll with a shorter timeout or something?

@rystsov - do you remember why `stopAll` hangs?

Reply:
After some discussion we decided to remove the `System.exit()` before `stopAll`: it is still there but after stopAll. @rystsov if this is a problem for your case, let's discuss it more when it occurs.